src/number_theory/number_field/embeddings.lean

```diff
@@ -123,3 +125,141 @@ end
+  mul_zero, tsub_zero, eq_self_iff_true, forall_const],
+  map_zero' := by simp only [map_zero, zero_re],
+  map_add' := by simp only [map_add, add_re, eq_self_iff_true, forall_const], }
+
+@[simp]
+lemma real_embedding_eq_embedding {φ : K →+* ℂ} (hφ : is_real φ) (x : K) :
+  (hφ.embedding x : ℂ) = φ x :=
+begin
+  ext, { refl, },
+  { rw [of_real_im, eq_comm, ← complex.eq_conj_iff_im],
+    rw is_real at hφ,
+    exact ring_hom.congr_fun hφ x, },
+end
+
+lemma place_real_embedding_eq_place {φ : K →+* ℂ} (hφ : is_real φ) :
+  place hφ.embedding = place φ :=
+by { ext x, simp only [place_apply, real.norm_eq_abs, ←abs_of_real, norm_eq_abs,
+  real_embedding_eq_embedding hφ x], }
+
+lemma is_real_conjugate_iff {φ : K →+* ℂ} :
+  is_real (conjugate φ) ↔ is_real φ := is_self_adjoint.star_iff
+
+end number_field.complex_embedding
+
+section infinite_place
+
+open number_field
+
+variables (K : Type*) [field K]
+
+/-- An infinite place of a number field `K` is a place associated to a complex embedding. -/
+def number_field.infinite_place := { w : absolute_value K ℝ  // ∃ φ : K →+* ℂ, place φ = w}
+
+instance [number_field K] : nonempty (number_field.infinite_place K) :=
+begin
+  rsuffices ⟨φ⟩ : nonempty (K →+* ℂ), { use ⟨place φ, ⟨φ, rfl⟩⟩, },
+  rw [← fintype.card_pos_iff, embeddings.card K ℂ],
+  exact finite_dimensional.finrank_pos,
+end
+
+variables {K}
+
+/-- Return the infinite place defined by a complex embedding `φ`. -/
+noncomputable def number_field.infinite_place.mk (φ : K →+* ℂ) : number_field.infinite_place K :=
+⟨place φ, ⟨φ, rfl⟩⟩
+
+namespace number_field.infinite_place
+
+open number_field
+
+instance : has_coe_to_fun (infinite_place K) (λ _, K → ℝ) := { coe := λ w, w.1 }
+
+lemma infinite_place_eq_place (φ : K →+* ℂ) (x : K) : (mk φ) x = (place φ) x := rfl
+
+lemma apply (φ : K →+* ℂ) (x : K) : (mk φ) x = complex.abs (φ x) := rfl
+
+/-- For an infinite place `w`, return an embedding `φ` such that `w = infinite_place φ` . -/
+noncomputable def embedding (w : infinite_place K) : K →+* ℂ := (w.2).some
+
+lemma infinite_place_embedding_eq_infinite_place (w : infinite_place K) :
+  mk (embedding w) = w :=
+by { ext, exact congr_fun (congr_arg coe_fn (w.2).some_spec) x, }
+
+lemma place_embedding_eq_infinite_place (w : infinite_place K) (x : K) :
+    place (embedding w) x = w x :=
+ begin
+   rw ← infinite_place_eq_place,
+   exact congr_fun (congr_arg coe_fn (infinite_place_embedding_eq_infinite_place w)) x,
+ end
+
+lemma nonneg (w : infinite_place K) (x : K) : 0 ≤ w x := w.1.nonneg _
+
+@[simp]
+lemma eq_zero (w : infinite_place K) (x : K)  : w x = 0 ↔ x = 0 := w.1.eq_zero
+
+@[simp]
+lemma map_zero (w : infinite_place K) : w 0 = 0 := w.1.map_zero
+
+@[simp]
+lemma map_one (w : infinite_place K) : w 1 = 1 := w.1.map_one
+
+@[simp]
+lemma map_mul (w : infinite_place K) (x y : K) : w (x * y) = (w x) * (w y) := w.1.map_mul _ _
+
+@[simp]
+lemma map_inv (w : infinite_place K) (x : K) : w (x⁻¹) = (w x)⁻¹ := map_inv₀ w.1 _
```

Comment:
If you provide a `monoid_with_zero_hom_class (infinite_place K) K R` instance then you don't need these lemmas.

Reply:
I am not sure it is possible since it requires to prove that `coe_injective` but two infinite places can define the same absolute value without being equals (ie corresponding to two different embeddings). The next PR in line #17285 is precisely about that.

Reply:
Well, scratch that. It is injective. Sorry for the noise.

Reply:
Ok, I added the instance and removed a bunch of unnecessary lemmas.